python/Ganga/GPIDev/Base/VPrinter.py

```diff
@@ -316,27 +320,28 @@ def summary_print(obj, out=None, interactive=False):
             print('[]', end=' ', file=out)
         else:
             outString = '['
-            count = 0
-            for x in _obj:
+            outStringList = []
+            for x in obj:
                 if isType(x, GangaObject):
-                    sio = StringIO()
+                    sio =StringIO()
                     stripProxy(x).printSummaryTree(0, 0, '', out=sio)
                     result = sio.getvalue()
                     # remove trailing whitespace and newlines
-                    outString += result.rstrip()
+                    outStringList.append(result.rstrip())
                 else:
-                    result = str(x)
                     # remove trailing whitespace and newlines
-                    outString += result.rstrip()
-                count += 1
-                if count != obj_len:
-                    outString += ', '
+                    outStringList.append(str(x).rstrip())
+                # Check if we're at the last element or not and add commas
+                if len(outStringList)+1 != 2*obj_len:
+                    outStringList.append(', ')
+            outString += ''.join(outStringList)
```

Comment:
Doesn't that give us an extra ',' at the start? Tests are running with the change now. If it causes problems I'm tempted to revert